---
title: Analysis of gene expression in colorectal cancer DLD1 cells with UHRF1 depletion
  and HDAC inhibition
author: "Mariona Alberola"
date: "22 de mayo de 2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
## Summary	
UHRF1 is a major regulator of epigenetic mechanism and is overexpressed in various human malignancies. In this study, we examined the involvement of UHRF1 in aberrant DNA methylation in colorectal cancer (CRC). In CRC cells, transient UHRF1 knockdown rapidly induced DNA demethylation across entire genomic regions, including CpG islands, gene bodies and repetitive elements. Nonetheless, UHRF1 depletion only minimally reversed CpG island hypermethylation-associated gene silencing. However, the combination of UHRF1 depletion and histone deacetylase (HDAC) inhibition synergistically reactivated the silenced genes and strongly suppressed CRC cell proliferation. Our results suggest that (i) maintenance of DNA methylation in CRC cells is highly dependent on UHRF1; (ii) UHRF1 depletion rapidly induces DNA demethylation, though it is insufficient to fully reactivate the silenced genes; and (iii) dual targeting of UHRF1 and HDAC may be an effective new therapeutic strategy.
 	
## Overall design	
DLD1 cells were transfected with a control siRNA (Thermo Fisher Scientific) or siRNAs targeting UHRF1 (Thermo Fisher Scientific) using using Lipofectamine RNAiMAX (Thermo Fisher Scientific) and incubated for 48 h. Cells were then treated with mock (ethanol) or 300 nM Trichostatin A (TSA) for additional 24 h.
 	
## Contributor(s)	
Suzuki H

## Citation(s)	
Niinuma T, Kitajima H, Kai M, Yamamoto E et al. UHRF1 depletion and HDAC inhibition reactivate epigenetically silenced genes in colorectal cancer cells. Clin Epigenetics 2019 May 7;11(1):70. PMID: 31064417






```{r}
targets<- read.csv("samples.txt", sep="")
targets

```


```{r}
expressions <- read.delim("GSE106402_series_matrix.txt", header = TRUE, row.names = 1)
class(expressions)
```


```{r}
dim(expressions)
summary(expressions)
```

```{r}
boxplot(log(expressions), col = c(1,2,3,4,5,6,7,8,9))
```
I used the log() function to normalize the data because it had many outliers. In the boxplot obtained we can see that the data is divided in two groups, one of them represented as boxplots and the other one represents the outliers after normalizing it.

```{r}

## ----simulateData--------------------------------------------------------
expressionValues <- data.matrix(expressions,rownames.force = NA)
```



```{r}
## ----simulateGeneInfo----------------------------------------------------
myGenes <-  rownames(expressionValues)
myGenes
```

```{r}

## ----simulateInfo--------------------------------------------------------
myInfo=list(myName="Mariona Alberola", myLab="Omics Lab", 
          myContact="mariona.alberola@alum.esci.upf.edu", myTitle="Practical Exercise on ExpressionSets")
show(myInfo)
```


```{r}
## ----loadPackage---------------------------------------------------------
require(Biobase)
```

Now we have to create the expression set:
```{r}
## ----creaExpressionSet1--------------------------------------------------
myEset <- ExpressionSet(expressionValues)
class(myEset)
show(myEset)
```


```{r}
## ----AnnotatedDataFrame2-------------------------------------------------
columnDesc <-  data.frame(labelDescription= c("Sample_Name", "Cell", "Experiment_Type", "TSA", "Replicate"))
myAnnotDF <- new("AnnotatedDataFrame", data=targets, varMetadata= columnDesc)
show(myAnnotDF)
```


```{r}
## ------------------------------------------------------------------------
phenoData(myEset) <- myAnnotDF
pData(myAnnotDF)

## ----eval=FALSE----------------------------------------------------------
## # (pData(myAnnotDF))<-pData(myAnnotDF)$sample_Name
## # Error in validObject(.Object) :
## #   invalid class ExpressionSet object: 1: sampleNames differ between assayData and phenoData
## # invalid class ExpressionSet object: 2: sampleNames differ between phenoData and protocolData
```


```{r}
## ------------------------------------------------------------------------
rownames(pData(myAnnotDF))<-pData(myAnnotDF)$Sample_Name
myEset <- ExpressionSet(assayData=expressionValues, phenoData=myAnnotDF)
show(myEset)
```


```{r}
## ------------------------------------------------------------------------
myEset <- ExpressionSet(assayData=expressionValues, 
                        phenoData=myAnnotDF, 
                        featureNames = myGenes)
# show(myEset)
```


```{r}
## ----label=MIAME---------------------------------------------------------
myDesc <- new("MIAME", name= myInfo[["myName"]],
            lab= myInfo[["myLab"]],
            contact= myInfo[["myContact"]] ,
            title=myInfo[["myTitle"]])
print(myDesc)
```


```{r}
## ------------------------------------------------------------------------
myEset <- ExpressionSet(assayData=expressionValues, 
                        phenoData=myAnnotDF,
                        fetureNames =myGenes,
                        experimentData = myDesc)
 show(myEset)
```

Now that the expression set has been created and we've already added all the necessary information, it's already prepared to be used
```{r}
## ----usingExpressionSets-------------------------------------------------
dim(exprs(myEset))
class(phenoData(myEset))
class(pData(phenoData(myEset)))
head(pData(phenoData(myEset)))
head(pData(myEset))
```


```{r}
## ------------------------------------------------------------------------
smallEset <- myEset[1:15,c(1:3,6:8)]
dim(exprs(smallEset))
dim(pData(smallEset))
head(pData(smallEset))
all(colnames(exprs(smallEset))==rownames(pData(smallEset)))
pData(smallEset)
```



```{r}
## ------------------------------------------------------------------------
if (!require(GEOquery)) {
  BiocManager::install("GEOquery")
}
require(GEOquery)
gse <- getGEO("GSE106402")
class(gse)
names(gse)
gse[[1]]
esetFromGEO <- gse[[1]]
esetFromGEO
```



